### PR TITLE
MINOR: Stateless transformation documentation

### PR DESCRIPTION
**Pull request by enothereska**
Needs to come after https://github.com/apache/kafka/pull/3701
Originally reviewed as part of #3490.

### CODE REVIEW

**Comment by enothereska** [comment]
cc @dguy

File: docs/streams/developer-guide.html [hunk 1]
```diff
@@ -466,7 +466,419 @@
     GlobalKTable&lt;String, GenericRecord&gt; source2 = builder.globalTable("topic4", "globalStoreName");
     </pre>
 
-    <h4><a id="streams_dsl_windowing" href="#streams_dsl_windowing">Windowing a stream</a></h4>
+    <h4><a id="streams_dsl_transform" href="#streams_dsl_transform">Transform a stream</a></h4>
+
+    <code>KStream</code> and <code>KTable</code> support a variety of transformation operations. Each of these operations can be translated into one or more
```
Comment (dguy) [inline docs/streams/developer-guide.html]: I know this is a bit annoying, but can we try and put each sentence on a new line? It just makes it easier to review in the future.

File: docs/streams/developer-guide.html [hunk 2]
```diff
@@ -466,7 +466,419 @@
...
+    connected processors into the underlying processor topology. Since <code>KStream</code> and <code>KTable</code> are strongly typed, all these transformation
+    operations are defined as generics functions where users could specify the input and output data types.
```
Comment (dguy) [inline docs/streams/developer-guide.html]: generics -> generic

File: docs/streams/developer-guide.html [hunk 3]
```diff
@@ -466,7 +466,419 @@
...
+
+    Some <code>KStream</code> transformations may generate one or more <code>KStream</code> objects (e.g., filter and map on <code>KStream</code> generate another <code>KStream</code>,
+    while branch on <code>KStream</code> can generate multiple <code>KStream</code>) while some others may generate a <code>KTable</code> object (e.g., aggregation) interpreted
```
Comment (dguy) [inline docs/streams/developer-guide.html]: multiple KStreams
 or 
multiple KStream Instances?

File: docs/streams/developer-guide.html [hunk 4]
```diff
@@ -466,7 +466,419 @@
...
+    as the changelog stream to the resulted relation. This allows Kafka Streams to continuously update the computed value upon arrivals
```
Comment (dguy) [inline docs/streams/developer-guide.html]: upon arrival

**Review by dguy — COMMENTED** [review]
Thanks @enothereska, looks like we are missing quite a lot of `<p>...</p>`

**Comment by enothereska** [comment]
Thanks @dguy

**Comment by enothereska** [comment]
2 unit tests failed: `org.apache.kafka.streams.integration.KStreamRepartitionJoinTest.shouldCorrectlyRepartitionOnJoinOperationsWithZeroSizedCache`
`org.apache.kafka.streams.integration.KStreamRepartitionJoinTest.shouldCorrectlyRepartitionOnJoinOperationsWithNonZeroSizedCache
`
Not related to PR obviously.

File: docs/streams/developer-guide.html [hunk 5]
```diff
@@ -571,7 +571,458 @@
         </tbody>
     </table>
 
-    <h4><a id="streams_dsl_windowing" href="#streams_dsl_windowing">Windowing a stream</a></h4>
+    <h4><a id="streams_dsl_transform" href="#streams_dsl_transform">Transform a stream</a></h4>
+    <p>
+    <code>KStream</code> and <code>KTable</code> support a variety of transformation operations. Each of these operations
+    can be translated into one or more connected processors into the underlying processor topology. Since <code>KStream</code>
+    and <code>KTable</code> are strongly typed, all these transformation operations are defined as generics functions where
```
Comment (dguy) [inline docs/streams/developer-guide.html:578]: generic functions

**Review by dguy — APPROVED** [review]
LGTM

**Comment by dguy** [comment]
Thanks @enothereska, merged to trunk

File: docs/streams/developer-guide.html (hunk 5 above)
Comment (enothereska) [reply docs/streams/developer-guide.html:578]: Getting this in next round, thanks!

**Pull request closed without merging**